src/api/app/services/scm_status_reporter.rb

```diff
@@ -7,51 +7,10 @@ def initialize(event_payload, event_subscription_payload, scm_token, event_type
     @state = event_type.nil? ? 'pending' : scm_final_state(event_type)
   end
 
-  # rubocop:disable Metrics/CyclomaticComplexity
-  # rubocop:disable Metrics/PerceivedComplexity
   def call
-    if github?
-      github_client = Octokit::Client.new(access_token: @scm_token, api_endpoint: @event_subscription_payload[:api_endpoint])
-      # https://docs.github.com/en/rest/reference/repos#create-a-commit-status
-      github_client.create_status(@event_subscription_payload[:target_repository_full_name],
-                                  @event_subscription_payload[:commit_sha],
-                                  @state,
-                                  status_options)
-    else
-      gitlab_client = Gitlab.client(endpoint: "#{@event_subscription_payload[:api_endpoint]}/api/v4",
-                                    private_token: @scm_token)
-      # https://docs.gitlab.com/ce/api/commits.html#post-the-build-status-to-a-commit
-      gitlab_client.update_commit_status(@event_subscription_payload[:project_id],
-                                         @event_subscription_payload[:commit_sha],
-                                         @state,
-                                         status_options)
-    end
-    @workflow_run.save_scm_report_success(request_context) if @workflow_run.present?
-  rescue Octokit::InvalidRepository => e
-    package = Package.find_by_project_and_name(@event_payload[:project], @event_payload[:package])
-    return if package.blank?
-
-    tokens = Token::Workflow.where(scm_token: @scm_token).pluck(:id)
-    return if tokens.none?
-
-    EventSubscription.where(channel: 'scm', token: tokens, package: package).delete_all
-
-    if @workflow_run.present?
-      @workflow_run.save_scm_report_failure("Failed to report back to GitHub: #{e.message}",
-                                            request_context)
-    end
-  rescue Octokit::Error, Gitlab::Error::Error => e
-    rescue_with_handler(e) || raise(e)
-  rescue Faraday::ConnectionFailed => e
-    if @workflow_run.present?
-      @workflow_run.save_scm_report_failure("Failed to report back to GitHub: #{e.message}",
-                                            request_context)
-    end
-  ensure
-    RabbitmqBus.send_to_bus('metrics', "scm_status_report,scm=#{@event_subscription_payload[:scm]},exception=#{e} value=1") if e.present?
+    # Should use either GithubStatusReport or GitlabStatusReport
+    raise NotImplementedError
```

Comment:
We should instead raise `AbstractMethodCalled`:

```suggestion
    raise AbstractMethodCalled
```